activerecord/lib/active_record/persistence.rb

```diff
@@ -139,7 +139,8 @@ def save(*)
     # Attributes marked as readonly are silently ignored if the record is
     # being updated.
     def save!(*)
-      create_or_update || raise(RecordNotSaved.new(nil, self))
+      create_or_update || raise(RecordNotSaved.new("Failed to save the record " \
+                                                   "because one of the before callbacks returned false", self))
```

Comment:
At master callbacks halt the execution using `throw(:abort)` we need to update the message now.